sam/record.go

```diff
@@ -105,9 +105,12 @@ func IsValidRecord(r *Record) bool {
 // Tag returns an Aux tag whose tag ID matches the first two bytes of tag and true.
 // If no tag matches, nil and false are returned.
 func (r *Record) Tag(tag []byte) (v Aux, ok bool) {
-	for i := range r.AuxFields {
-		if bytes.Compare(r.AuxFields[i][:2], tag) == 0 {
-			return r.AuxFields[i], true
+	if len(tag) != 2 {
```

Comment:
This should test `len(tag) < 2`. I'm not sure that we should return instead of panic here. A panic was the previous behaviour, though we could be more informative.